### PR TITLE
Added README.md for composition

### PR DESCRIPTION
**Pull request by cardboardcode**
## Purpose of Pull Request

This pull request adds a `README.md` to **composition** as part of efforts to address the issue highlighted in https://github.com/ros2/demos/issues/531.

### CODE REVIEW

File: composition/README.md [hunk 1]
```diff
@@ -147,9 +147,12 @@ When executed correctly, strings should be printed to terminal similar to what i
 
 ## **FAQ**
 
-WIP
+`Q`: Why use Composition?
+
+`A`: Composition provides utilties to  manage life cycle for nodes. This provides greater control over the state of a given ROS-based system. The idea of a managed node is to provide a interactable interface which executes to a known life cycle state machine, allowing for higher system transparency and code-agnosticism.
```
Comment (mjeronimo) [inline composition/README.md:152]: This section is describing lifecyle/managed nodes, not node composition. Node composition allows nodes to reside in the same process where the overhead of marshaling and unmarshaling messages can be avoided.

**Review by mjeronimo — CHANGES_REQUESTED** [review]

File: composition/README.md (hunk 1 above)
Comment (cardboardcode) [reply composition/README.md]: Noted. Updated the section to the following:

```md
`Q`: Why use node composition?

`A`: Node composition avoids the overhead of marshaling and unmarshalling messages by allowing nodes to be instantiated within the same process.
```

Comment (cardboardcode) [reply composition/README.md]: Done.

**Review by audrow — COMMENTED** [review]
Looks pretty good to me. I've added a few small fixes.

File: composition/README.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
+## **What Is This?**
+
+This demo provides examples on 3 different ways which users could use the **rclcpp_components** API to compose multiple nodes in a single process.
```
[the remark this answers is not in the document]
Comment (cardboardcode) [reply composition/README.md:3]: Done.

File: composition/README.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+## **Run**
+
+### Manual Composition
+
+Running `manual_composition` compiles an executable that runs the following 4 components: 
+
+- **Talker** (A ROS 2 component which publishes a string)
+- **Listener** (A ROS 2 component that prints the received string from **Talker**)
+- **Server** (A ROS 2 component that adds two integers and outputs its result to **Client**)
+- **Client** (A ROS 2 component that sends two integers to **Server** and prints the received result from **Server**).
```
[the remark this answers is not in the document]
Comment (cardboardcode) [reply composition/README.md:28]: Noted. Done.

File: composition/README.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
+## **What Is This?**
```
[the remark this answers is not in the document]
Comment (cardboardcode) [reply composition/README.md]: Noted. Removed all bolding from all headings.

**Comment by mjeronimo** [comment]
Yes, I'll take a look...

File: composition/README.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
+## What Is This?
+
+This demo provides examples of three different ways to use the **rclcpp_components** API to compose multiple nodes in a single process.
+
+This ROS 2 package consists of the following nodes:
```
Comment (mjeronimo) [inline composition/README.md:5]:
```suggestion
This ROS 2 package consists of the following demo applications:
```

**Review by mjeronimo — APPROVED** [review]
Approved with one minor suggestion.

File: composition/README.md (hunk 5 above)
Comment (cardboardcode) [reply composition/README.md]: Done.

**Pull request merged**